lib/rust/ensogl/core/src/display/world/stats.rs

```diff
@@ -44,28 +50,42 @@ impl {
             monitor.add::<debug::monitor::SpriteSystemCount>(),
             monitor.add::<debug::monitor::SpriteCount>(),
         ];
-        Self {stats,performance,monitor,panels}
+        Self {stats,are_stats_valid,performance,monitor,panels}
     }
 
     /// Start measuring data.
-    pub fn begin(&mut self) {
-        if self.visible() {
+    /// Returns a snapshot of statistics data for the previous rendering frame if the monitor is
+    /// [`visible()`] (and if it was visible during the previous frame as well).
+    pub fn begin(&mut self) -> Option<StatsData> {
```

Comment:
Why function that "Returns a snapshot of statistics data for the previous rendering frame ..." is called "begin"? For me that does not make sense now and I do not understand the behavior/semantics of this function anymore.